src/interface.jl

```diff
@@ -2,83 +2,101 @@
 
 
 """
-    isdensitytype(type::Type)
+    hasdensity(d)::Bool
```

Comment:
Why get rid of the Type method? These are sometimes important for generated functions

Reply:
We discussed that on #1. Initially I had thought using the type would be better. But I had to concede that using the instance was better:

> I thought it would be better to decide based on the type, so it can be checked if you don't have an object in hand. But maybe isdensity(density::SomeDensity) would be better: I'm currently adding a function get_device() to KernelAbstractions (JuliaGPU/KernelAbstractions.jl#269), and I originally implemented it to decide based on array type. But Valentin Churavy asked me to change it and use the array instance, he pointed out "One of the learnings for me from Adapt.jl is that you don't want to work on types, working on instances is more reliable."

Reply:
I agree the primary method should takes values instead of types. I sometimes include methods for both, when it makes sense. The main concern here is, is there a way this interface can easily be used for generated functions?

Reply:
You can use `hasdensity` in the code generated by the generated function :smile: 

I strongly suggest not adding a type-based definition for now since there is no use case yet (in contrast to the instance-based definition which @oschulz already uses in some code). Additionally, generally it is recommended to either only use instances or types to avoid confusion (and instances are recommended): https://docs.julialang.org/en/v1/manual/style-guide/index.html#Avoid-confusion-about-whether-something-is-an-instance-or-a-type

Reply:
From your link:
> The preferred style is to use instances by default, and only add methods involving `Type{MyType}` later if they become necessary to solve some problems.

I agree with this. If we need this for generated functions, I guess we can just add methods later since MeasureTheory owns the measure types :)

Reply:
I'll mark this resolved, then.